eclipsesource/theia
# Add `task#runOptions` to API.

#### What it does

Additionally added functionality to support `reevaluateOnRerun`. If this flag is set to false no variables will be resolved again. Therefore lastTask can now be re-triggered without resolving all variables. For this the lastTask object was updated to contain all resolved information.

Contributed on behalf of STMicroelectronics

Resolves # 11132
#### How to test

To test you can use the follwing `tasks.json`:
```
{
  "version": "2.0.0",
  "tasks": [
    {
      "label": "Node Command",
      "type": "shell",
      "command": "node",
      "args": [
        "${input:node command}"
      ],
      "isBackground": true,
      "runOptions": {
        "reevaluateOnRerun": false
      }
    },
  ],
  "inputs": [
    {
      "id": "node command",
      "description": "node ${input:node command}",
      "default": "--version",
      "type": "promptString"
    },
  ]
}
```

When you run the node command you will be asked, which command you want to run. If you enter `--version` for example and then rerun the last task. You will not be asked and `node --version` is automatically executed.
When you change `reevaluateOnRerun` to true, run and rerun the task, you will need to reenter the command every time.

To test the API you can use this [extension](https://github.com/eclipse-theia/theia/files/9764231/task-provider-samples-0.0.1.zip). It provides a task (custombuildscript: 32) with `reevaluateOnRerun: true`. When you run this task and debug, for example at `task-service.ts#808` the `task` object should have the correct `runOptions` set.
#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
